### PR TITLE
Removed `PintaCore` references from `DocumentLayers`

### PR DESCRIPTION
**Pull request by Lehonti**
The creation of the `DocumentLayers` instance happens when the global environment is still getting initialized, so it could be hard to ascertain if things are initialized in the right order (i.e. if the `PintaCore.Layers` and `PintaCore.Tools` in question are defined at the point where they are used).

In this case, I added a few `Console.WriteLine` statements in my test environment to test this, and both services are, indeed, defined at that point. This is a stronger guarantee than knowing they are initialized at the point they are first used.

### CODE REVIEW

**Comment by cameronwhite** [comment]
These dependencies are questionable so I think it would be better to see if we can avoid them in the first place - `Document` and `DocumentLayers` really should be at the leaf of the dependency tree and not depend on higher level things

- Instead of passing a global layer manager into every document, `LayerManager` could handle everything itself - it just needs to listen for an event like `ActiveDocumentChanged` and then could forward the per-document events like `LayerAdded` to its own events
- Or, eliminate `LayerManager` entirely. It's only used by the text tool  which could have the same changes as above to listen for the active document's layer events
- The `tools.Commit()` could probably be moved up to the layers list widget, since this was mainly intended to have tools like the text tool and shape tool commit their in-progress edits before switching layers.

**Comment by Lehonti** [comment]
I completely agree that the dependencies are questionable @cameronwhite 

I eliminated `LayerManager` and now all of its events are part of `WorkspaceManager`. Very importantly, `Document` doesn't depend on them, but rather the `WorkspaceManager` attaches handlers to `Document` events.

As for the dependency on `ToolManager`, I think that removing it could be a bit involved, and this pull request is already quite large, so what if we leave this for a future one?

File: Pinta.Core/Managers/WorkspaceManager.cs [hunk 1]
```diff
@@ -259,6 +291,10 @@ public void CloseDocument (
 			}
 		}
 
+		document.Layers.LayerAdded += Document_LayerAdded;
```
Comment (cameronwhite) [inline Pinta.Core/Managers/WorkspaceManager.cs:294]: Should these have been `-=` to remove the events?

**Review by cameronwhite — COMMENTED** [review]
Yeah, leaving the tools change for another commit is wise since that'll need more testing.

File: Pinta.Core/Managers/WorkspaceManager.cs (hunk 1 above)
Comment (Lehonti) [reply Pinta.Core/Managers/WorkspaceManager.cs]: Yup @cameronwhite , my mistake. I just corrected it

**Pull request merged**